CHANGELOG.md

```diff
@@ -7,12 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.4.0] - 2024-04-20
+
 ### Changed
 
+Version 0.4.0 comes with a number of small breaking changes which we've decided to release all as part of a single version. More breaking changes in one release is inconvenient, but we've tried to coordinate them in hopes that any future breaking changes will be non-existent or very rare. All changes will get picked up the Go compiler, and each one should be quite easy to fix. The changes don't apply to any of the most common core APIs, and likely many projects won't have to change any code.
```

Comment:
Should we put a ⚠️ emoji or anything with this? It’s a rare scenario and we want it to be extra visible.

Reply:
Done.